core/extensions/extension.hpp

```diff
@@ -0,0 +1,140 @@
+    virtual void ext_print_num(uint64_t value) = 0;
+    virtual void ext_print_utf8(const uint8_t *utf8_data,
+                                uint32_t utf8_length) = 0;
+
+    /// cryptographic extensions
+    virtual void ext_blake2_256(const uint8_t *data,
+                                uint32_t len,
+                                uint8_t *out) = 0;
+    virtual void ext_blake2_256_enumerated_trie_root(const uint8_t *values_data,
+                                                     const uint32_t *lens_data,
+                                                     uint32_t lens_length,
+                                                     uint8_t *result) = 0;
+    virtual uint32_t ext_ed25519_verify(const uint8_t *msg_data,
+                                        uint32_t msg_len,
+                                        const uint8_t *sig_data,
+                                        const uint8_t *pubkey_data) = 0;
+    virtual void ext_twox_128(const uint8_t *data,
+                              uint32_t len,
+                              uint8_t *out) = 0;
+    virtual void ext_twox_256(const uint8_t *data,
+                              uint32_t len,
+                              uint8_t *out) = 0;
+
+    /// sandboxing extensions
+    virtual void ext_sandbox_instance_teardown(uint32_t instance_idx) = 0;
+    virtual uint32_t ext_sandbox_instantiate(
+        std::function<uint64_t(const uint8_t *serialized_args,
+                               uint32_t serialized_args_length,
+                               uint32_t state,
+                               uint32_t func_index)> dispatch_func,
+        const uint8_t *wasm_ptr,
+        uint32_t wasm_length,
+        const uint8_t *imports_ptr,
+        uint32_t imports_length,
+        uint32_t state) = 0;
+    virtual uint32_t ext_sandbox_invoke(uint32_t instance_idx,
+                                        const uint8_t *export_ptr,
+                                        uint32_t export_len,
+                                        const uint8_t *args_ptr,
+                                        uint32_t args_len,
+                                        uint8_t *return_val_ptr,
+                                        uint32_t return_val_len,
+                                        uint32_t state) = 0;
+    virtual uint32_t ext_sandbox_memory_get(uint32_t memory_idx,
+                                            uint32_t offset,
+                                            uint8_t *buf_ptr,
+                                            uint32_t buf_length) = 0;
+    virtual uint32_t ext_sandbox_memory_new(uint32_t initial,
+                                            uint32_t maximum) = 0;
+    virtual uint32_t ext_sandbox_memory_set(uint32_t memory_idx,
+                                            uint32_t offset,
+                                            const uint8_t *val_ptr,
+                                            uint32_t val_len) = 0;
```

Comment:
Here and in other places where in rust source code usize type is used.
```pub fn ext_sandbox_memory_set(
			memory_idx: u32,
			offset: u32,
			val_ptr: *const u8,
			val_len: usize,
		) -> u32;```

"The size of this primitive is how many bytes it takes to reference any location in memory.
For example, on a 32 bit target, this is 4 bytes and on a 64 bit target, this is 8 bytes."
https://doc.rust-lang.org/std/primitive.usize.html
It may be important to know what number of bytes returns webassembly.